### DISCUSSION TITLE
Got 404 in calling the Server extension backend endpoint on Jupyterlite

### DISCUSSION THREAD

**Discussion by DaWingChan** [jupyterlite/jupyterlite#1298] [Q&A]
**System**
`Windows,
jupyterlite-core:0.2.3,
jupyterlab:4.0.11, 
python:3.9, 
nodejs:18`

Hi Team,

I encountered an issue on working the lab server extension with Jupyterlite. 
After `pip` installed the extension on my local disk and rebuild the Jupyterlite wihtout any issue, I can see the extension can be loaded in the console log.
`pre_build:federated_extensions:copy:ext:@jupyterlab-examples/server-extension`
```
federated_extensions:copy:ext:@jupyterlab-examples/server-extension
federated_extensions:copy:ext:@jupyterlite/pyodide-kernel-extension
-- pre_build:federated_extensions:copy:ext:jupyterlab_pygments
-- pre_build:federated_extensions:copy:ext:@jupyterlab-examples/server-extension
``` 
However a 404 error occurred on the browser while calling the extension backend handler's endpoint.

![jupyterlite_error](https://github.com/jupyterlite/jupyterlite/assets/29755808/b46708f5-935f-4292-aed9-96fc53ad6b49)

As the `cookiecutter` is deprecated, I use the `copier` and follow the guideline in building the lab server extension from [JupyterLab Server Extension Example](https://github.com/jupyterlab/extension-examples/tree/main/server-extension), without any code changes, by below commands
Both `.tar.gz` and `.whl` are built successfully
```
pip install -e .
jupyter labextension develop . --overwrite
jupyter server extension enable jupyterlab_examples_server
jlpm run build
```

The extension is work when I run `jupyter lab`, response comes back while calling the handler's endpoint
![JupyterLab](https://github.com/jupyterlite/jupyterlite/assets/29755808/daf08d2a-3b3b-4926-a7aa-07c41b512412)

Also I have double checked the server extension config under the etc/ in the the .venv, the extension is enabled.
```
{
  "ServerApp": {
    "jpserver_extensions": {
      "jupyterlab_examples_server": true
    }
  }
}
```
 
Here is my repo: [Jupyterlite-poc](https://github.com/DaWingChan/jupyterlite-poc), the built packages are under packages dir

May I know how can I make it work under the JupyterLite

Best regards

**Comment by Linnore** [comment]
Hi, any update on this issue? I am also interested in adding customized endpoints to the jupyterlite.

**Comment by jtpio** [comment]
This is "expected" behavior since there is no server in JupyterLite. JupyterLite implements a small "server" in the browser to handle kernels and other endpoints used by the JupyterLab frontend.

If you want to make extensions that will work in JupyterLite, maybe you could look into making a JupyterLab extension: https://jupyterlite.readthedocs.io/en/stable/howto/extensions/frontend.html

**Reply by Linnore** [reply]
I went through the example you provided in the docs. 

Unfortunately, if we want to create customized endpoints, then that would correspond to the server extension instead of the frontend extension. The current example of server extension in the docs is deprecated since the template is not updated to JupyterLab 4＋.
 https://github.com/jupyterlite/serverlite-cookiecutter-ts

Will JupyterLite team consider update the cookiecutter or provide an updated example of creating customized endpoints?

Thanks!
